packages/lib/src/spatial-navigation/components/virtualizedList/getNumberOfItemsVisibleOnScreen.ts

```diff
@@ -0,0 +1,28 @@
+const getMinSizeOfItems = <T>(data: T[], itemSize: number | ((item: T) => number)) => {
+  if (data.length === 0) {
+    throw new Error('The data array is empty.');
+  }
+
+  if (typeof itemSize === 'number') {
+    return itemSize;
+  }
+
+  let minSize = itemSize(data[0]); // Initialize with the size of the first item
+
+  for (let i = 1; i < data.length; i++) {
+    const currentSize = itemSize(data[i]);
+    if (currentSize < minSize) {
+      minSize = currentSize;
+    }
+  }
+
+  return minSize;
+};
+export const getNumberOfItemsVisibleOnScreen = <T>(
+  data: T[],
+  listSizeInPx: number,
+  itemSize: number | ((item: T) => number),
+) => {
+  const itemSizeToComputeRanges = getMinSizeOfItems(data, itemSize);
```

Comment:
C'est plus un minimum number of items visible on screen non ? On se sert de cette props pour le jump on scroll, est ce qu'on a pas un cas où ca peut casser le scroll ? (Si on a un petit element et plusieurs gros par exemple)

Reply:
![image](https://github.com/user-attachments/assets/eb949341-a191-4f56-b68f-bc345a21c3c3)
On peut pas faire du jump on scroll avec des tailles dynamiques :D C'est pour ça que je me suis permis de faire ça